pywhy_graphs/algorithms/generic.py

```diff
@@ -333,3 +334,192 @@ def _single_shortest_path_early_stop(G, firstlevel, paths, cutoff, join, valid_p
+
+    return out
+
+
+def _find_directed_children(G, node):
+    """Finds the children of a node in the directed and the bidirected subgraphs.
+
+    Args:
+        G : Graph
+            The graph.
+        node : node label
+            The node for which we have to find the children.
+
+    Returns:
+        out : set
+            The children of the provided node.
+    """
+    if not isinstance(G, CPDAG):
+        bidirected_children = set(G.sub_bidirected_graph().neighbors(node))
+    directed_children = set(G.sub_directed_graph().successors(node))
+
+    out = bidirected_children.union(directed_children)
+
+    return out
+
+
+def _is_collider(G, node):
+    """Checks if the given node is a collider or not.
+
+    Args:
+        G : graph
+            The graph.
+        node : node
+            The node to be checked.
+
+    Returns:
+        iscollider : bool
+            Bool is set true if the node is a collider, false otherwise.
+    """
+    parents = _find_directed_parents(G, node)
+
+    if len(parents) > 1:
+        return True
+    return False
+
+
+def _recursive_path(G, node_x, node_y, L, S, visited, xyancestors, cur_node):
+    """Recursively explores a graph to find a path.
+
+       Finds path that are compliant with the inducing path requirements.
+
+    Args:
+        G : graph
+            The graph.
+        node_x : node
+            The source node.
+        node_y : node
+            The destination node
+        L : set
+            Set containing all the non-colliders.
+        S : set
+            Set containing all the colliders.
+        visited : set
+            Set containing all the nodes already visited.
+        xyancestors : set
+            Set containing the ancestors of X and Y.
+        cur_node : node
+            The current node.
+
+    Returns:
+        path : Tuple[bool, path]
+            A tuple containing a bool and a path if the bool is true.
+    """
+    path_exists = False
+    path = []
+    visited.add(cur_node)
+    children = _find_directed_children(G, cur_node)
+
+    if cur_node is node_y:
+        return (True, [node_y])
+
+    for elem in children:
+        if elem in visited:
+            continue
+        else:
+            print(_is_collider(G, elem))
+            if (
+                _is_collider(G, elem)
+                and (elem not in S)
+                and (elem not in xyancestors)
+                and (elem is not node_y)
+            ):
+                continue
+            elif not _is_collider(G, elem) and (elem not in L) and (elem is not node_y):
+                continue
+            path_exists, temp_path = _recursive_path(
+                G, node_x, node_y, L, S, visited, xyancestors, elem
+            )
+            if path_exists:
+                path.append(cur_node)
+                path.extend(temp_path)
+                break
+    return (path_exists, path)
+
+
+def inducing_path(G, node_x, node_y, L=set(), S=set()):
```

Comment:
You should not initialize to a set(), or any mutable object. https://stackoverflow.com/questions/1534407/python-object-initialization-bug-or-am-i-misunderstanding-how-objects-work

A pattern is initializing to `None` and then using if/else conditionals to set the default parameters. You should note this in the docstring also.